carmin-server/server/resources/helpers/executions.py

```diff
@@ -1,10 +1,138 @@
 import os
 import json
+from typing import Dict
+from server import app
+from server.database.models.user import User
+from server.database.models.execution import Execution as ExecutionDB
+from server.resources.models.error_code_and_message import ErrorCodeAndMessage
+from server.resources.models.pipeline import Pipeline, PipelineSchema
+from server.common.error_codes_and_messages import (
+    UNAUTHORIZED, INVALID_INPUT_FILE, INVALID_PATH, EXECUTION_NOT_FOUND,
+    INVALID_PIPELINE_IDENTIFIER, EXECUTION_IDENTIFIER_MUST_NOT_BE_SET,
+    UNEXPECTED_ERROR, ErrorCodeAndMessageFormatter)
 from server.resources.models.execution import Execution
+from .path import create_directory, get_user_data_directory, is_safe_path, is_data_accessible, platform_path_exists
 
+INPUTS_FILENAME = "inputs.json"
+EXECUTIONS_DIRNAME = "executions"
 
-def write_inputs_to_file(execution: Execution, path_to_execution_dir: str):
-    inputs_json_file = os.path.join(path_to_execution_dir, 'inputs.json')
+
+def create_execution_directory(execution: ExecutionDB,
+                               user: User) -> (str, ErrorCodeAndMessage):
+
+    user_execution_dir = os.path.join(
+        get_user_data_directory(user.username), EXECUTIONS_DIRNAME)
+    try:
+        os.mkdir(user_execution_dir)
+    except FileExistsError:
+        pass
+    execution_dir_absolute_path = os.path.join(user_execution_dir,
+                                               execution.identifier)
+
+    if not is_safe_path(execution_dir_absolute_path) or not is_data_accessible(
+            execution_dir_absolute_path, user):
+        return None, UNAUTHORIZED
+
+    path, error = create_directory(execution_dir_absolute_path)
+    return execution_dir_absolute_path, error
+
+
+def get_execution_dir(username: str, execution_identifier: str) -> str:
```

Comment:
Maybe we could add a possible error here that would be raised if the folder does not exist, or we can keep this as a simple way to join the path information.

Reply:
Yes, I see this more as a simple join helper function that does not check if the `pipeline_identifier` is valid. That work would be left to the caller. Also, the function is not currently being used, so it might even not be necessary.

Reply:
The function now raises a `FileNotFoundError` is the directory does not exist.